src/main/java/software/amazon/cloudwatchlogs/emf/model/Histogram.java

```diff
@@ -0,0 +1,147 @@
+/*   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package software.amazon.cloudwatchlogs.emf.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import software.amazon.cloudwatchlogs.emf.Constants;
+import software.amazon.cloudwatchlogs.emf.exception.InvalidMetricException;
+
+/** Histogram metric type */
+class Histogram extends Statistics {
+    Histogram(List<Double> values, List<Integer> counts) {
+        if (counts.size() != values.size()) {
+            throw new IllegalArgumentException("Counts and values must have the same size");
```

Comment:
Just wondering why in this case we use `IllegalArgumentException` but on similar constraint validation below we throw `InvalidMetricException` instead?